include/bluetooth/services/ots.h

```diff
@@ -0,0 +1,600 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_BLUETOOTH_SERVICES_OTS_H_
+#define ZEPHYR_INCLUDE_BLUETOOTH_SERVICES_OTS_H_
+
+/**
+ * @brief Object Transfer Service (OTS)
+ * @defgroup bt_gatt_ots Object Transfer Service (OTS)
+ * @ingroup bluetooth
+ * @{
+ * @brief API for the Object Transfer Service (OTS).
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <zephyr/types.h>
+#include <sys/util.h>
+#include <bluetooth/conn.h>
+#include <bluetooth/uuid.h>
+
+/** @brief Type of an OTS object. */
+struct bt_gatt_ots_obj_type {
```

Comment:
Other services has already been added with the `bt_gatt` prefix, so I think we should continue to do so.

One thing I think we should resolve is how to distinguish between server and client versions. of the services.
At the moment we only have 3 services, HRS, DIS and BAS, all of them are server models.

In Bluetooth Mesh the models have the `_cli_` and `_srv_` infixes in the name, should we adopt this for the bluetooth services as well?

Reply:
Good question, because there's a client version of OTS (dubbed OTC) coming, and if we want to start added _srv_ or _cli_ we might as well start now (unless we are just goint with ots/otc and bas/bac, etc.)

Reply:
> unless we are just goint with ots/otc and bas/bac, etc

The `s` is for service though. I would suggest `_ots_srv_` and `_ots_cli_` similar to Bluetooth Mesh models.

Reply:
That's true. Depending on the profile, the naming convention is sometimes e.g. [Feature] Server and [Feature] Client, although in this case, the the names given in the profile spec is just Object Transfer Service and Object Client. 

But I would be fine with the _srv_ and _cli_ naming convention.

Reply:
Would it then be ``bt_ots_srv_`` or ``bt_gatt_ots_srv_``? I prefer the first option because it is shorter.

Do you want me to align the naming in this PR?

Reply:
@kapi-no I think it would probably be better to settle on a naming convention before you update all the code :) 

I agree with the argument you have with the shorter name, but on the other hand I also see a reason why to prefix it with `gatt`. I don't have any strong feelings between the two naming prefix (`bt_ots_srv_` and `bt_gatt_ots_srv_`); both works for me and I can't find anything in the documentation that strictly forbids either.

Reply:
> @kapi-no I think it would probably be better to settle on a naming convention before you update all the code :)
> 

Yes, that's why I am asking about both variants (so that we agree on the naming).


>Do you want me to align the naming in this PR?
>

I probably wasn't precise enough in this question. Do you want me to align the naming in this PR or should it rather be a part of another PR which aligns naming for all the service libraries in Zephyr (bas, dis, ...)?

Reply:
Ah, misunderstood you then :) I think we should include whatever naming scheme we settle on in the PR, rather than updating it later.